### PR TITLE
feat: add auto-cors

### PR DESCRIPTION
**Pull request by Apoorva64**
This PR adds dynamic cors handeling to microcks rest endpoints.
It takes the request made to microcks and returns the cors headers generated with the request's origin header.
If that header is not found it uses a default value.

Testing:
Go to https://microcks.io/ in your browser. Do a fetch request in the console to a mock

### CODE REVIEW

File: webapp/src/main/java/io/github/microcks/web/filter/DynamicOriginCorsFilter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package io.github.microcks.web.filter;
```
Comment (lbroudoux) [inline webapp/src/main/java/io/github/microcks/web/filter/DynamicOriginCorsFilter.java:1]: The copyright header is required here

**Review by lbroudoux — CHANGES_REQUESTED** [review]
Looks good to me but code need to be reformatted using 4 spaces as tab.
(Sorry, I know we have an ongoing dicussion on this ;-) )
Let me know if you can quickly push a change otherwise I'll merge and reformat after that

**Comment by Apoorva64** [comment]
> Looks good to me but code need to be reformatted using 4 spaces as tab. (Sorry, I know we have an ongoing dicussion on this ;-) ) Let me know if you can quickly push a change otherwise I'll merge and reformat after that

Sry about the formatting. I forgot to set it to 4 spaces. I don't currently easily modify this branch(it's on another PC). Could you also rename the 'corsFilter2' to an appopriate name

**Comment by lbroudoux** [comment]
No worry. I'm gonna merge it and re-format afterwards. Thanks a lot for the contribution!

**Pull request merged**